crates/router/src/core/disputes/transformers.rs

```diff
@@ -0,0 +1,106 @@
+use common_utils::errors::CustomResult;
+
+use crate::{
+    core::{errors, files::transformers::retrieve_file_and_provider_file_id_from_file_id},
+    routes::AppState,
+    types::SubmitEvidenceRequestData,
+};
+
+pub async fn get_evidence_request_data(
+    state: &AppState,
+    merchant_account: &storage_models::merchant_account::MerchantAccount,
+    evidence_request: api_models::disputes::SubmitEvidenceRequest,
+    dispute: &storage_models::dispute::Dispute,
+) -> CustomResult<SubmitEvidenceRequestData, errors::ApiErrorResponse> {
+    let (cancellation_policy, cancellation_policy_provider_file_id) =
+        retrieve_file_and_provider_file_id_from_file_id(
```

Comment:
Can you please explain about why this function is being called so many times, and each time the resulting variables seem different.